src/webextension/manage/containers/currentEntry.js

```diff
@@ -20,25 +46,40 @@ function CurrentEntry({noEntry, entry, onSave, onDelete}) {
     );
   }
   return (
-    <EntryDetails entry={entry} onSave={onSave} onDelete={onDelete}/>
+    <EntryDetails entry={flattenEntry(entry)} onSave={onSave}
+                  onDelete={onDelete}/>
   );
 }
 
+CurrentEntry.propTypes = {
+  noEntry: PropTypes.bool.isRequired,
+  entry: PropTypes.shape({
+    id: PropTypes.string.isRequired,
+    title: PropTypes.string.isRequired,
+    entry: PropTypes.shape({
+      username: PropTypes.string.isRequired,
+      password: PropTypes.string.isRequired,
+    }),
+  }),
+  onSave: PropTypes.func.isRequired,
+  onDelete: PropTypes.func.isRequired,
+};
+
 CurrentEntry = connect(
   (state) => {
+    if (state.ui.newEntry)
+      return { noEntry: false, entry: null };
     return {
-      noEntry: !state.ui.newEntry && state.ui.selectedEntry === null,
-      entry: state.ui.newEntry ? null : state.storage.entries.find(
-        (x) => x.id === state.ui.selectedEntry
-      )
+      noEntry: state.cache.currentEntry === null,
+      entry: state.cache.currentEntry,
     };
   },
   (dispatch) => {
     return {
     onSave: (entry) => {
       if (entry.id === undefined)
```

Comment:
in the future, let's make sure every conditional is a block (avoids having a GOTO FAIL style error)

Reply:
Resolved with an eslint check to ensure we don't forget.